Add NGC containers to Kubeflow

supertetelman opened the pull request:
Still be tested/cleaned up.

This adds a script that will update the Kubeflow config.yaml file and other templates files to include NGC containers, etc.

Future updates will include NGC style changes.

michael-balint commented on scripts/k8s_deploy_kubeflow.sh:
```diff
@@ -76,6 +79,10 @@ fi
 cd ${KFAPP}
+
+# Update the Kubeflow Jupyter UI
+./scripts/update_kubeflow_config.py
```
Why do we do this twice?

supertetelman commented:
I added a missing line that was causing some failures and removed the now unnecessary extra script execution. I tested this a few times and it is now populating with NGC containers.

There are still two improvements we should probably make to this. These shouldn't be blocking for this PR and instead are future improvements, but let me know what you think.

1. Only show the first 3-10 tags for each container (currently the list is very long
2. Pre-populate the extra-resources with nvidia.com/gpu values instead of giving an example in the help message

supertetelman commented:
I added a missing line that was causing some failures and removed the now unnecessary extra script execution. I tested this a few times and it is now populating with NGC containers.

There are still two improvements we should probably make to this. These shouldn't be blocking for this PR and instead are future improvements, but let me know what you think.

1. Only show the first 3-10 tags for each container (currently the list is very long
2. Pre-populate the extra-resources with nvidia.com/gpu values instead of giving an example in the help message

supertetelman commented:
Does thee `kfctl.sh apply k8s` command not update the config map when it runs `kubectl apply --validate=false -f default.yaml`? I guess I assumed that running that would properly update Kubeflow. I'll dig into that a bit more and see if that is the issue.

I'll also push a few additional fixes/enhancements in here now based on other feedback.

michael-balint commented:
@supertetelman - No, running `/opt/kubeflow/scripts/kfctl.sh apply k8s` doesn't appear to update. Still get the old images.
